src/NativeScript/ObjC/Constructor/ObjCConstructorBase.mm

```diff
@@ -102,6 +103,8 @@ static void writeAdapter(ExecState* execState, const JSValue& value, void* buffe
         this->_ffiTypeMethodTable.write = &writeAdapter<TNSArrayAdapter>;
     } else if (klass == [NSDictionary class]) {
         this->_ffiTypeMethodTable.write = &writeAdapter<TNSDictionaryAdapter>;
+    } else if (klass == [NSData class]) {
```

Comment:
Maybe we should make this checks with `isSubclassOfClass` for the adapters, what do you think?